### PR TITLE
Add The Watcher.app v1.0

### PR DESCRIPTION
**Pull request by ibips**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/the-watcher.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  name "The Watcher"
+  desc "Monitor ETH, Gas, & Wallets"
+  homepage "https://watcher.app/"
+
+  livecheck do
+    url "https://watcher.app/downloads/latest-version.txt"
+    regex(/([0-9.]+)/)
+  end
+
+  app "The Watcher.app"
```
Comment (neutric) [inline Casks/the-watcher.rb:15]: Please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

File: Casks/the-watcher.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  url "https://watcher.app/downloads/TheWatcher_v#{version}.zip"
+  name "The Watcher"
+  desc "Monitor ETH, Gas, & Wallets"
+  homepage "https://watcher.app/"
+
+  livecheck do
+    url "https://watcher.app/downloads/latest-version.txt"
+    regex(/([0-9.]+)/)
+  end
+
```
Comment (neutric) [inline Casks/the-watcher.rb:14]: Please add `depends_on macos: ">= :monterey"` (source: `homepage`).

File: Casks/the-watcher.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "3a38bb76f147a377f4ea25983bcf5bf6a883d08cc3a75d3317c2aba65fbe6bf6"
+
+  url "https://watcher.app/downloads/TheWatcher_v#{version}.zip"
+  name "The Watcher"
+  desc "Monitor ETH, Gas, & Wallets"
+  homepage "https://watcher.app/"
+
+  livecheck do
+    url "https://watcher.app/downloads/latest-version.txt"
+    regex(/([0-9.]+)/)
```
Comment (miccal) [inline Casks/the-watcher.rb:12]:
```suggestion
    regex(/(\d+(?:\.\d+)+)/)
```

File: Casks/the-watcher.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+cask "the-watcher" do
+  version "1.0"
+  sha256 "3a38bb76f147a377f4ea25983bcf5bf6a883d08cc3a75d3317c2aba65fbe6bf6"
+
+  url "https://watcher.app/downloads/TheWatcher_v#{version}.zip"
+  name "The Watcher"
+  desc "Monitor ETH, Gas, & Wallets"
```
Comment (miccal) [inline Casks/the-watcher.rb:7]:
```suggestion
  desc "Monitor ETH wallets, USD-ETH prices and gas prices"
```

File: Casks/the-watcher.rb (hunk 3 above)
Comment (neutric) [inline Casks/the-watcher.rb:12]: Please use the more structured go-to regex for version matching: `regex(/v?(\d+(?:\.\d+)+)/)`

**Review by neutric — COMMENTED** [review]
Thanks for your contribution to `homebrew-cask`, @ibips! My review is in the code. Take it with a grain of salt as I am not a maintainer.

**Comment by ibips** [comment]
@neutric 

- added Monterey dependency with https://github.com/Homebrew/homebrew-cask/pull/117418/commits/4f57569dad8ae191d7880cd9f7714feb5fe723e1

- I use CreateZap script to generate the zap stanza, and it doesn't seem it found anything that needs to be removed. 
  ```
  % brew createzap the-watcher
  No the-watcher settings found.
  ```
 - thanks for the review!

**Comment by neutric** [comment]
> I use CreateZap script to generate the zap stanza, and it doesn't seem it found anything that needs to be removed.

It is highly unlikely that `CreateZap` finds nothing, given the right search term. Try searching for `watcher` instead of `the-watcher`. Also, you have to have started the app at least once so it can create these files, but that is probably not the problem. If `CreateZap` can't find anything, please use `AppCleaner` (`brew install appcleaner`) to identify the right search term and feed that to `CreateZap`.

**Comment by ibips** [comment]
@neutric the appcleaner did the trick. Thanks!

File: Casks/the-watcher.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  depends_on macos: ">= :monterey"
+
+  app "The Watcher.app"
+
+  zap trash: [
+    "~/Library/Application Scripts/io.galactic.web3console",
+    "~/Library/Containers/io.galactic.web3console",
+    "~/Library/Group Containers/2VV9TDDQG6.group.io.galactic.web3console",
+    "/var/db/receipts/io.galactic.web3console.*",
```
Comment (miccal) [inline Casks/the-watcher.rb:23]: Please remove `"/var/db/receipts/io.galactic.web3console.*"` from the `zap` -- files in `/var` are temporary files that should not be removed by `uninstall` or `zap`.

**Comment by miccal** [comment]
Thank you @ibips.

**Pull request merged**